go/arrow/array/concat.go

```diff
@@ -363,15 +364,22 @@ func sumArraySizes(data []arrow.ArrayData) int {
 	return outSize
 }
 
+func getListViewOffsets[T int32 | int64](data arrow.ArrayData, i int) []T {
```

Comment:
since you're using this for both offsets and sizes, I'd suggest changing the name just to avoid possible confusion.

Reply:
Going with `getListViewBufferValues[T]`. This will soon be migrated to an API inspired by what bkietz recently added to C++ now that Go also has "templates".